### ISSUE TITLE
Add fusion graph tests

### ISSUE THREAD

**Issue by sgligorijevicTT** [tenstorrent/tt-xla#3577]
Initial batch of patterns that will get unit tests:

Patterns from `TTIRFusing.cpp`:
- [x] ConvAddBias 
- [x] ReductionWithReshapePattern
- [x] SoftmaxFusionPattern 
- [x] NumericStableSoftmaxFusionPattern
- [x] ReluFusionPattern 
- [x] Relu6FusionPattern 
- [x] HardsigmoidFusionPattern 
- [x] SiluFusionPattern 
- [x] MishFusingPattern 
- [x] ConvWithMultiply 
- [ ] ConvTagWeights (won't test, doesn't replace op with another op, only adds an attr, that might end up as a different layout on other things which is too hard to test)
- [x] BatchNormDecomposition
- [x] RepVGGConvSumFusionPattern 
- [x] MatmulWithBiasFusionPattern 
- [x] ScaledSumToMeanPattern 
- [x] SpatialMeanOptimizationPattern 
- [x] ConcatenateHeadsUpdatePattern 
- [x] SplitQueryKeyValueAndSplitHeadsUpdatePattern
- [x] GeluFusionPattern 
- [x] RMSNormFusionPattern 
- [x] PermuteMatmulFusionPattern 
- [x] ReshapeBroadcastReshapeToRepeatPattern

Patterns from `TTNNFusing.cpp`:
- [x] TTNNConv2dWithActivation
- [x] TTNNMatmulAndLinearWithActivation
- [x] SDPAFusing

Issue text will be updated as patterns get tests added and if the the issue is rescoped

Scope was slightly expended to add rudimentary filechecks to most important models, and add some separate graph/"semi-integration" tests
